src/trie.js

```diff
@@ -82,30 +82,31 @@ var marexandre;
     Trie.prototype.getIndices = function(_text_) {
       var self = this;
       var result = [];
-      var copy = '';
-      var tmpTrie = self.list;
+      var remainingText = '';
+      var currentNode = self.list;
       var start = -1, end = -1;
 
       for (var i = 0, imax = _text_.length; i < imax; i++) {
-        copy = _text_.slice(i);
+        remainingText = _text_.slice(i);
 
         // TODO: Need to refactor this loop :(
-        for (var j = 0, jmax = copy.length; j < jmax; j++) {
-          var c = copy[j];
-          var exists = tmpTrie.children.hasOwnProperty(c.toString());
+        for (var j = 0, jmax = remainingText.length; j < jmax; j++) {
+          var c = remainingText[j];
 
-          if (exists) {
-            tmpTrie = tmpTrie.children[c];
+          if (self.hasChildrenWithValue(currentNode, c)) {
```

Comment:
in addition, i don't want to write old style javascript class implementation. it might be good chance to take `ES2016` and babel compiler.